crates/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -47,8 +48,10 @@ pub struct ExecutionContext<'vm, 'state, 'a, 'b, E: fmt::Debug, S: StorageView<E
     pub tx_context: &'a mut TxContext,
     /// The gas status used for metering
     pub gas_status: &'a mut SuiGasStatus<'b>,
-    /// The session used for interacting with Move types and calls
-    pub session: Session<'state, 'vm, S>,
+    /// The native extension to be passed to the VM at execution time.
+    /// It wraps the state view and most importantly the `ObjectRuntime` that
+    /// the execution is going to use.
+    pub object_runtime_extension: NativeContextExtensions<'state>,
```

Comment:
this is the "big" change and something that I think moves in the right direction. Maybe some fields could be combined into a single abstraction (some data store, runtime concept), maybe not.
@tnowacki will talk about his